miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppSecureStorageDispatcher.kt

```diff
@@ -0,0 +1,169 @@
+        storageState = state
+    }
+
+    fun setBridgeExecutor(activity: Activity, bridgeExecutor: MiniAppBridgeExecutor) {
+        this.activity = activity
+        this.bridgeExecutor = bridgeExecutor
+    }
+
+    fun setMiniAppComponents(miniAppId: String) {
+        this.miniAppId = miniAppId
+        this.secureStorage = MiniAppSecureStorage(activity)
+        secureStorage.storageState.observeForever(stateObserver)
+        onLoad()
+    }
+
+    @Suppress("ComplexCondition")
+    private fun <T> whenReady(callback: () -> T) {
+        if (this::bridgeExecutor.isInitialized &&
+            this::activity.isInitialized &&
+            this::miniAppId.isInitialized &&
+            this::secureStorage.isInitialized
+        ) {
+            callback.invoke()
+        }
+    }
+
+    private fun onLoad() = whenReady {
+        val onSuccess = { items: Map<String, String> ->
+            cachedItems = items
+            bridgeExecutor.dispatchEvent(
+                eventType = NativeEventType.MINIAPP_SECURE_STORAGE_READY.value,
+                value = Gson().toJson(SecureStorageReadyCallback(true))
+            )
+        }
+        val onFailed = { error: MiniAppStorageError ->
+            bridgeExecutor.dispatchEvent(
+                eventType = NativeEventType.MINIAPP_SECURE_STORAGE_READY.value,
+                value = Gson().toJson(SecureStorageReadyCallback(false, error))
+            )
+        }
+        secureStorage.loadSecureStorage(miniAppId, onSuccess, onFailed)
+    }
+
+    fun onSetItems(callbackId: String, jsonStr: String) = whenReady {
+        val callbackObj: SecureStorageCallbackObj? =
+            Gson().fromJson(jsonStr, SecureStorageCallbackObj::class.java)
+        if (callbackObj != null) {
+            val onSuccess = { items: Map<String, String> ->
+                cachedItems = items
+                bridgeExecutor.postValue(callbackId, SAVE_SUCCESS_SECURE_STORAGE)
+            }
+            val onFailed = { error: MiniAppStorageError ->
+                bridgeExecutor.postError(callbackId, Gson().toJson(error))
+            }
+            if (storageState != StorageState.LOCK)
+                secureStorage.insertSecureStorageItem(
+                    miniAppId,
+                    callbackObj.param,
+                    onSuccess,
+                    onFailed
+                )
+            else
+                bridgeExecutor.postError(
+                    callbackId,
+                    Gson().toJson(MiniAppStorageError.storageOccupiedError)
+                )
+        } else {
+            bridgeExecutor.postError(callbackId, ERR_WRONG_JSON_FORMAT)
+        }
+    }
+
+    fun onGetItem(callbackId: String, jsonStr: String) = whenReady {
+        val callbackObj: GetItemCallbackObj? =
+            Gson().fromJson(jsonStr, GetItemCallbackObj::class.java)
+        if (callbackObj != null) {
+            if (!cachedItems.isNullOrEmpty() && cachedItems!!.containsKey(callbackObj.param)) {
```

Comment:
In the case that the key doesn't exist, I think we should just return `null` instead of an error. So maybe you can add a specific case for `!cachedItems!!.containsKey(callbackObj.param)`.

Also, maybe we should avoid using `!!` and use a null check instead?

Reply:
@munir-rakuten See this comment again - I think iOS updated the behavior to return `null` so Android needs to match.

Reply:
@corycaywood addressed